### PR TITLE
FT: Users Bookmark Accommodations

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

- Adds a bookmark.js file responsible for 

#### Description of Task to be completed?
- Allow a user to bookmark an accommodation
- Allow a user to remove a bookmarked accommodation
- Allow a user to view their bookmarks

#### How should this be manually tested?
- Clone this repository
- Run ` npm run install `
- Run` npm run dev`
- Log in as a user
- Navigate to the` accommodaitons list ` page
- Hit the `Star` icon
- VIew `My Bookmarks` page
#### Any background context you want to provide?
N/ A
#### Screenshots (if appropriate)

Screen Shot 2020-01-07 at 13 04 26
Screen Shot 2020-01-07 at 13 04 51
Screen Shot 2020-01-07 at 13 05 05
Screen Shot 2020-01-07 at 13 05 24

#### Questions:
N / A

### CODE REVIEW

**Review by alainmateso — CHANGES_REQUESTED** [review]
Good job @Cheza-Dzabala! However, I believe when viewing the bookmarks image section should be of the same size on each card to make it look better. Also putting the view accommodation button at the bottom of the card would look nicer since now they don't seem to align well depending on different content.

File: src/actions/bookmarkActions.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export const getBookmarks = () => async (dispatch) => {
+  try {
+    const response = await backendCall.get('/bookmarks', { headers });
+    dispatch(bookmarksType(GET_BOOKMARKS, response.data));
+  } catch (error) {
+    dispatch(bookmarksType(GET_BOOKMARKS_ERROR, error.response));
+  }
+};
+
+
```
Comment (nakiwuge) [inline src/actions/bookmarkActions.js]: Remove this space

File: src/components/pages/accommodations/BookmarkStar.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-nested-ternary */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Star, StarBorderOutlined } from '@material-ui/icons';
+import { bookmarkAccommodation } from '../../../actions/bookmarkActions';
+
```
Comment (nakiwuge) [inline src/components/pages/accommodations/BookmarkStar.js]: Remove this space

File: src/components/pages/accommodations/BookmarkStar.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+class BookmarkStar extends Component {
```
Comment (nakiwuge) [inline src/components/pages/accommodations/BookmarkStar.js]: Rename this class to `BookMark`

File: src/components/pages/accommodations/BookmarkStar.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        break;
+      default:
+        this.setState({
+          bookmarking: true,
+        });
+    }
+  }
+
+  processBookmark = async () => {
+    const { bookmarkAccommodation, slug } = this.props;
```
Comment (nakiwuge) [inline src/components/pages/accommodations/BookmarkStar.js]: Add a new line

File: src/components/pages/accommodations/BookmarkStar.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    await bookmarkAccommodation(slug);
+    this.setState({
+      processedBookmarking: !this.state.processedBookmarking,
```
Comment (nakiwuge) [inline src/components/pages/accommodations/BookmarkStar.js]: destructure the state

File: src/views/Bookmarks.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  showBookmarks = () => {
+    const { bookmarks } = this.props;
+    return (
+      (bookmarks.data && bookmarks.data.length > 0)
+        ? (
+          <Container style={{ padding: '10px ' }} fluid>
+            <Row className="bookmark-row">
+              {bookmarks.data.map((bookmark) => (
+                <Card style={{ width: '18rem' }}>
+                  <Card.Img variant="top" src={(typeof (bookmark.accommodation.images) === 'string') ? bookmark.accommodation.images : bookmark.accommodation.images[0]} />
```
Comment (nakiwuge) [inline src/views/Bookmarks.js]: destructure accommodation from bookmark

**Review by alainmateso — APPROVED** [review]

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job. Kindly work on the feedback below.
- A user should be able to remove a bookmark from the list of bookmarks on the `my bookmarks` page.
- Give the bookmarks page a heading
- Give the images of the accommodations on `my bookmarks` page the same size
- Capitalize the accommodations tittles
- Reduce the space between the location icon and location
- Add some space below available rooms
- Reduce the space below the `View Accommodations button`
Screenshot 2020-01-07 at 15 19 50

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Great job so far @Cheza-Dzabala 
However, I suggest you return the right currency on the accommodation which is apprently not the case as you can see in the screenshot below. I had noticed the same in all accommodations or single accommodation. It would be great if you could correct that as well.
👍 

Screen Shot 2020-01-07 at 17 25 38

**Review by nignanthomas — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**